### ISSUE TITLE
Python-redmine export all columns

### ISSUE THREAD

**Issue by giladste** [maxtepkeev/python-redmine#221]
Now that the export function is implemented, I would like to ask how can I export ALL columns, including custom fields. The following code exports only the mandatory predefined fields, neglecting all my custom fields which are not listed:

> report = redmine.issue.filter(project_id=PROJECT_ID,
> tracker_id=TRACKER_ID,
> status_id=OPEN_STATUS,
> cf_12=CPE_FW_ver) 
> .export(fmt='csv', savepath='.', filename=OUTPUT_FILENAME)

When exporting using the web-UI, it is possible to check the box on "All columns".
I did not see any help on the subject over at https://python-redmine.com/resources/custom_field.html
Would love some help on how to apply the check box using the redmine-python project.
Thanks

**Comment by maxtepkeev** [comment]
Hi @giladste 

Can you please be more specific regarding the "All columns" checkbox, because I can't seem to find it anywhere. A screenshot would help.

Thanks.

**Comment by giladste** [comment]
![image](https://user-images.githubusercontent.com/20501527/53740235-70373f80-3e9c-11e9-94e4-278418b8a6fc.png)

**Comment by maxtepkeev** [comment]
Ah, I see, thanks, looks like it only works for CSV export, have never noticed this dialog box before :(

Currently there's no way to control which columns will be exported and which not, so we'll need to make some code modifications to make it work.

Thanks for raising this issue, that will be a nice feature to have, will start work on it ASAP, unfortunately can't tell you the specific ETA for now.

**Comment by giladste** [comment]
Thanks max, this would be put into great use for me.
Can I get a "grosso-modo" estimation? Is it something I can wait for in the next 1-2 months or should I go manually rather than waiting?
Sorry to nag but I need to evaluate my future steps in the task
Thanks in advance and good luck

**Comment by maxtepkeev** [comment]
@giladste I'll do my best to implement this functionality until the end of the month.

**Comment by maxtepkeev** [comment]
@giladste As I'm not sure that I'll have time to implement this feature until the end of the month as promised I suggest you to use the following workaround for now:

```python
from redminelib import Redmine, utilities, exceptions


def export(resourceset, fmt, savepath=None, filename=None):
    if resourceset.manager.resource_class.query_all_export is None:
        raise exceptions.ExportNotSupported

    formatter = utilities.MemorizeFormatter()

    url = resourceset.manager.redmine.url + formatter.format(
        resourceset.manager.resource_class.query_all_export, format=fmt, **resourceset.manager.params)

    params = dict({'c[]': 'all_inline', 'encoding': 'UTF-8'}, **formatter.unused_kwargs)

    try:
        return resourceset.manager.redmine.download(url, savepath, filename, params=params)
    except exceptions.UnknownError as e:
        if e.status_code == 406:
            raise exceptions.ExportFormatNotSupportedError
        raise e

redmine = Redmine('http://redmine.url', username='*****', password='*****')

issues = redmine.issue.all()
print export(issues, 'csv', savepath='/home/giladste', filename='issues.csv')
```

**Comment by maxtepkeev** [comment]
Implemented in 137ea84a025a86ec7277a96bc3fa7c10b429f6fa
